node/core/pvf/src/error.rs

```diff
@@ -81,32 +118,18 @@ pub enum InvalidCandidate {
 impl From<PrepareError> for ValidationError {
 	fn from(error: PrepareError) -> Self {
 		// Here we need to classify the errors into two errors: deterministic and non-deterministic.
+		// See [`PrepareError::is_deterministic`].
 		//
-		// Non-deterministic errors can happen spuriously. Typically, they occur due to resource
-		// starvation, e.g. under heavy load or memory pressure. Those errors are typically transient
-		// but may persist e.g. if the node is run by overwhelmingly underpowered machine.
-		//
-		// Deterministic errors should trigger reliably. Those errors depend on the PVF itself and
-		// the sc-executor/wasmtime logic.
-		//
-		// For now, at least until the PVF pre-checking lands, the deterministic errors will be
-		// treated as `InvalidCandidate`. Should those occur they could potentially trigger disputes.
+		// We treat the deterministic errors as `InvalidCandidate`. Should those occur they could
+		// potentially trigger disputes.
 		//
+		// TODO: Is this up-to-date?
 		// All non-deterministic errors are qualified as `InternalError`s and will not trigger
 		// disputes.
```

Comment:
I wanted to check if this comment is up-to-date. Part of the point of adding preparation retries was to cut down on disputes for non-deterministic errors, but seems like that was unnecessary?

Reply:
Looks correct to me, even if we get non-deterministic error several times it still shouldn't trigger a dispute.

Reply:
Good question on whether this is up2date. What does git blame say - how old is this comment?

Reply:
Got it, wanted to double-check. @eskimor: https://github.com/paritytech/polkadot/commit/c10f0619379d0042c311224a012d5cc630aae478